### DISCUSSION TITLE
External Scaler: Define scale count based on condition

### DISCUSSION THREAD

**Discussion by spalaka** [kedacore/keda#2027] [Q&A / Need Help]
Hi Experts,

I am using an external scaler to scale the consumer based on the queue size.
https://keda.sh/docs/1.4/concepts/external-scalers/#built-in-scalers-interface

But the challenge is when IsActive returns true, it only scales one pod (In my case Queue consumer).
Is there a way to scale more pods based on a condition?

Ex: 
If queue depth maintains less than 1000 messages, scaling one pod is good enough to process the message.
If queue depth is more than 1000, 2 pods should scale
if greater than 10,000 msgs, should scale to 5 pods

Regards,
Suresh Palaka

**Comment by tomkerkhove** [comment]
We no longer support v1.4, is it possible to upgrade to 2.x?

https://github.com/kedacore/keda/blob/main/SECURITY.md#supported-versions

**Reply by spalaka** [reply]
Hi,

As per my deployment yaml, KEDA is at 2.2 version.

Pod Template:
  Labels:           app=keda-operator
  Service Account:  keda-operator
  Containers:
   keda-operator:
    Image:      ghcr.io/kedacore/keda:2.2.0
    Port:       8080/TCP
    Host Port:  0/TCP

I have pasted the wrong URL with the 1.4 version.

Here is the right URL to which I am referring:
https://keda.sh/docs/2.2/concepts/external-scalers/

Regards,
Suresh Palaka

**Reply by zroubalik** [reply]
Isn't this something that you could tweak in your scaler's implementation? Here you can see how HPA calculates the desired replica count based on the provided metrics: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details
